src/transformStateWithClones.js

```diff
@@ -1,13 +1,61 @@
 'use strict';
 
-/**
- * @param {Object} state
- * @param {Object[]} actions
- *
- * @return {Object[]}
- */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const compArr = [];
+  const compObj = { ...state };
+  const result = [];
+
+  result.push(state);
+
+  for (let i = 0; i < actions.length; i++) {
+    const objOfActions = actions[i];
+    const objForWorkKey = objOfActions.type;
+    const objForWorkValue = objOfActions.extraData
+    || objOfActions.keysToRemove;
+    const objForWork = {};
+
+    objForWork[objForWorkKey] = objForWorkValue;
+    compArr.push(objForWork);
+  }
+
+  for (let i = 0; i < compArr.length; i++) {
+    const el = compArr[i];
+    const controlKey = Object.keys(el)[0];
+
+    switch (controlKey) {
+      case 'clear':
```

Comment:
```suggestion
      case CLEAR:
```
It's better to use constants instead of magic strings